src/vivarium/component.py

```diff
@@ -16,7 +16,6 @@
 from datetime import datetime, timedelta
 from importlib import import_module
 from inspect import signature
-from numbers import Number
```

Comment:
I probably would have created a Number type that is `int | float` to minimize the changes needed. Since you've already done this it's probably fine, but _might_ be worth looking through the rest of the code base to see if we are importing Number from numbers elsewhere

Reply:
I already looked; this PR removes every instance of `from numbers import Number`